server/configuration.go

```diff
@@ -96,6 +100,14 @@ func (p *Plugin) setConfiguration(configuration *configuration) {
 		panic("setConfiguration called with the existing configuration")
 	}
 
+	// PluginDirectory & ProfileImageURL should be set based on server configuration and not the plugin configuration
+	if serverConfiguration.PluginSettings.Directory != nil {
+		configuration.PluginsDirectory = *serverConfiguration.PluginSettings.Directory
+	}
+	if serverConfiguration.ServiceSettings.SiteURL != nil {
+		configuration.ProfileImageURL = path.Join(*serverConfiguration.ServiceSettings.SiteURL, "/plugins/github/assets/profile.png")
```

Comment:
You could make this even for flexible, if you use something like in https://github.com/mattermost/mattermost-plugin-github/blob/68f61d01f1f4d23a95562cdb6075b90aa556fc57/server/api.go#L217.